internal/bus/chainsubscriber.go

```diff
@@ -275,42 +264,50 @@ func (s *chainSubscriber) revertChainUpdate(tx sql.ChainUpdateTx, cru chain.Reve
 	// NOTE: host updates are not reverted
 
 	// v1 contracts
-	var cus []contractUpdate
-	cru.ForEachFileContractElement(func(fce types.FileContractElement, _ bool, rev *types.FileContractElement, resolved, valid bool) {
-		cus = append(cus, v1ContractUpdate(fce, rev, resolved, valid))
+	var err error
+	cru.ForEachFileContractElement(func(fce types.FileContractElement, created bool, rev *types.FileContractElement, resolved, _ bool) {
+		if err != nil {
```

Comment:
Technically correct and probably the best way to do it indeed, but it's still a bit of a code smell we have to weirdly use this `err` to essentially `continue` for the remainder of the `ForEachX`.  We used to collect updates for that reason. I think the `ForEachX` methods should return an error and take a function that returns an error and break on error...

Reply:
I think we had that discussion in the past and decided to keep this interface. If you want to can open an issue on core to discuss changing it but I'm not sure it's bad enough to break the interface again.